prereise/gather/griddata/hifld/data_process/profiles.py

```diff
@@ -100,3 +104,98 @@ def build_wind(wind_plants, download_directory, year):
+    :param int/str year: year of data to fetch and build profiles for.
+    :param dict hydro_kwargs: keyword arguments to pass to
+        :func:`prereise.gather.hydrodata.eia.fetch_historical.get_generation`.
+    :return: (*pandas.DataFrame*) -- data frame of normalized power profiles. The index
+        is hourly timestamps for the profile year, the columns are plant IDs, the values
+        are floats.
+    """
+    # Build shapes for each region
+    query_regions = set(hydro_plants["Balancing Authority Code"].unique())
+    query_regions.add("US48")  # total U.S. hydro profile, to be used as a fall-back
+    hydro_kwargs["year"] = year
+    generation = get_generation(eia_api_key, regions=query_regions, **hydro_kwargs)
+    matching_regions = set(generation.columns) - {"US48"}
+
+    # Filter generation anomalies based on installed capacity
+    ba_groups = hydro_plants.groupby("Balancing Authority Code")
+    ba_capacities = ba_groups["Pmax"].sum()
+    # Initial screen: any points > 2x capacity or < -1 * capacity
+    generation.loc[:, matching_regions] = generation[matching_regions].apply(
+        lambda x: filter_anomalies_impute(x, capacity=ba_capacities.loc[x.name])
+    )
+
+    # Normalize regional totals by regional capacities (or peak generation)
+    normalized_by_generation = generation[matching_regions].columns[
+        generation[matching_regions].max() > ba_capacities.loc[matching_regions]
+    ]
+    if len(normalized_by_generation) > 0:
+        print("regions normalized by generation profile:")
+        print(
+            generation[normalized_by_generation].max()
+            / ba_capacities.loc[normalized_by_generation]
+        )
+    normalizing_capacity = pd.concat(
+        [ba_capacities.loc[matching_regions], generation[matching_regions].max()],
+        axis=1,
+    ).max(axis=1)
+    normalized_profiles = generation[matching_regions] / normalizing_capacity
+
+    # Build shape for 'default' (weighted average of other shapes)
+    normalized_profiles["default"] = (
+        normalized_profiles * ba_capacities.loc[normalized_profiles.columns]
+    ).sum(axis=1) / ba_capacities.loc[normalized_profiles.columns].sum()
+
+    # Build the output dataframe, and populate it
+    profiles = pd.DataFrame(
+        index=pd.date_range(start=f"{year}-01-01-00", end=f"{year}-12-31-23", freq="H"),
+        columns=hydro_plants.index,
+    )
+    for region in query_regions - {"US48"}:
+        plant_ids = ba_groups.get_group(region).index
+        if region in normalized_profiles:
+            template = normalized_profiles[region]
+        else:
+            print(region, "uses 'default' hydro profile")
+            template = normalized_profiles["default"]
+        # pandas won't broadcast for us, so we need to construct a 2D object
+        profiles.loc[:, plant_ids] = pd.concat(
+            [template] * len(plant_ids), axis=1
+        ).values
+
+    return profiles
```

Comment:
The implementation of the method looks fine. The assumption that all generators have the same hourly profile may be questioned by users. The load disaggregation has the the similar issue. I will post the question into our slack channel.